src/main/CommandLine.cpp

```diff
@@ -1041,30 +1041,109 @@ runReportLastHistoryCheckpoint(CommandLineArgs const& args)
+            duration = std::stoul(input.substr(separatorIndex + 1), &pos);
+        }
+        catch (std::exception&)
+        {
+            throw std::runtime_error(
+                "simulate-sleep-per-op-{slow, medium, fast} must be of the "
+                "form percentage/duration");
+        }
+    };
+
+    uint32_t percentage, duration, totalPercentage{0};
+    std::vector<std::pair<uint32_t, uint32_t>> ret;
+    for (auto const& input : inputs)
+    {
+        parser(input, percentage, duration);
+        if (!ret.empty() && ret.back().second < duration)
+        {
+            throw std::runtime_error(
+                "duration for slow, medium, fast must be non-increasing "
+                "(i.e., slow >= medium >= fast)");
+        }
+        ret.push_back(std::make_pair(percentage, duration));
+        totalPercentage += percentage;
+    }
+    if (totalPercentage != 100)
+    {
+        throw std::runtime_error(
+            "The sum of the percentages in "
+            "--simulate-apply-per-op-{slow, medium, fast} must equal 100");
+    }
+    for (auto const& p : ret)
+    {
+        LOG_INFO(DEFAULT_LOG, "Sleeps for {} usec {}\% of the time", p.second,
+                 p.first);
+    }
+    return ret;
+}
+
 int
 run(CommandLineArgs const& args)
 {
     CommandLine::ConfigOption configOption;
     auto disableBucketGC = false;
-    uint32_t simulateSleepPerOp = 0;
     std::string stream;
     bool inMemory = false;
     bool waitForConsensus = false;
     uint32_t startAtLedger = 0;
     std::string startAtHash;
+    std::string simulateApplyPerOpSlow, simulateApplyPerOpMedium,
+        simulateApplyPerOpFast;
+
+    auto simulateParserGenerator = [](std::string const& speed) {
+        return [&speed](std::string& simulateApplyPerOpListInput) {
+            return clara::Opt{
+                simulateApplyPerOpListInput,
+                "PERCENTAGE_AND_USEC"}["--simulate-apply-per-op-" + speed](
+                "Simulate application time per operation in the form of "
+                "percentage/duration");
+        };
+    };
 
-    auto simulateParser = [](uint32_t& simulateSleepPerOp) {
-        return clara::Opt{simulateSleepPerOp,
+    uint32_t simulateApplyPerOp = 0;
+    auto simulateParser = [](uint32_t& simulateApplyPerOp) {
+        return clara::Opt{simulateApplyPerOp,
                           "MICROSECONDS"}["--simulate-apply-per-op"](
-            "simulate application time per operation");
+            "deprecated: Use --simulate-apply-per-op-{fast, medium, slow} "
+            "instead");
     };
 
     return runWithHelp(
         args,
         {configurationParser(configOption),
          disableBucketGCParser(disableBucketGC),
-         simulateParser(simulateSleepPerOp), metadataOutputStreamParser(stream),
-         inMemoryParser(inMemory), waitForConsensusParser(waitForConsensus),
+         simulateParser(simulateApplyPerOp),
+         simulateParserGenerator("fast")(simulateApplyPerOpFast),
```

Comment:
there is no need to hardcode slow/medium/high:
`clara::Opt` can take a vector as a parameter, so in this case you would add a parser like
```
std::vector<std::string> perOpPairs;
clara::Opt{confs, "PERCENT/MICROSECOND"}["--simulate-apply-per-op-dist"]("percent followed by time in usecond (specified several times, sum of percent must be 100)")}
```

Reply:
Updated it. Thanks for this idea! This is much cleaner :)